DOCUMENTATION.md

```diff
@@ -1,810 +1,807 @@
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Invalid token: signature, api token not found"`<br>`}]`|Use the correct API key. Refer to [API keys](/#api-keys) for more information|
+|`404`|`[{`<br>`"error": "NoResultFound",`<br>`"message": "No Result Found"`<br>`}]`|Check your [template ID](/#get-a-template-by-id-and-version-arguments-template-id-required) and [version](/#version)|
 
 ## Get all templates
 
-#### Method
+### Method
 
-This will return the latest version of each template for your service.
-
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+This will return the latest version of all templates inside a collection named `templates`.
 
 ```ruby
-args = {
-  'template_type' => 'sms'
+args: {
+  'type' => 'sms'
 }
-templates = client.get_all_templates(args)
+templates: client.get_all_templates(args)
 ```
 
+### Arguments
 
-</details>
+#### type (optional)
 
+If you do not use `type`, the client returns all templates. Otherwise you can filter by:
 
-#### Response
+- `email`
+- `sms`
+- `letter`
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+### Response
+
+If the request to the client is successful, the client will return a `Notifications::Client::TemplateCollection`:
 
 ```ruby
-    TemplateCollection templates;
+TemplateCollection "TemplateCollectionName";
 ```
-If the response is successful, a TemplateCollection is returned.
 
-If no templates exist for a template type or there no templates for a service, the templates list will be empty.
+You must then run `TemplateCollectionName.collection` to return an array of the required templates.
 
-Otherwise the client will raise a `Notifications::Client::RequestError`.
+A template will take this format:
 
-</details>
+_NEED EXAMPLE_
 
+If no templates exist for a template type or there no templates for a service, the templates list will be empty.
 
-#### Arguments
+_NEED EXAMPLE_
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+### Error codes
 
-##### `template_type`
-If omitted all templates are returned. Otherwise you can filter by:
+If the request is not successful, the client will return a `Notifications::Client::RequestError` containing the relevant error code:
 
-* `email`
-* `sms`
-* `letter`
+|error.code|error.message|How to fix|
+|:---|:---|:---|
+|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "Template type is not one of [sms, email, letter]"`<br>`}]`|Contact the Notify team|
+
+## Generate a preview template
 
-</details>
+### Method
 
+This will generate a preview version of a template.
 
-## Generate a preview template
+```ruby
+template_preview: client.generate_template_preview(id, personalisation{})
+```
 
-#### Method
+The parameters in the personalisation argument must match the placeholder fields in the actual template. The API notification client will ignore any extra fields in the method.
```

Comment:
While it's true that you could put some unnecessary fields into `personalisation` by mistake and everything would work fine, I don't think we need to mention this.